### ISSUE TITLE
[Feature] Pseudo Proxy

### ISSUE THREAD

**Issue by nbrugger-tgm** [nbrugger-tgm/reactj#31]
In order to use generic and automatic reactive Objects you need to use `ReactiveProxy<M>`.

The problem is that you always have to use `ReactiveProxy<MyObject>` instead of `MyObject`, which is a little inconvenient

**Comment by nbrugger-tgm** [comment]
## implemented

The changes are now implemented.

### Changes
1. You can now make your data class implement `ProxySubject` (markup interface, no methods needed)
2. "Old style" proxies will now be created using `ReactiveProxy.createProxy`
3. "New style" proxies will now be created using `ReactiveProxy.create`

**Comment by nbrugger-tgm** [comment]
# [BUG] Proxy subject methods not forwarded correctly

When binding a proxy subject (properly created) to a UI changes in the UI don't reflect in the object/subject

this is due to a wong condition in `ProxySubject`  
```java
if (thisMethod.getDeclaringClass().equals(ProxySubject.class)) {
            return this.forwardProxySubjectCallToMyself(thisMethod, args);
}
```
this will allways return `false` as the declaring class is still `Reactable`

**Comment by nbrugger-tgm** [comment]
## Reproduced

The bug was reporduced and is now covered by a testcase.

The previous explanation was only partial right.
```java
if (thisMethod.getDeclaringClass().equals(ProxySubject.class)) {
            return this.forwardProxySubjectCallToMyself(thisMethod, args);
}
```
worked for the most part as only one method slips through.
This is due to `ReactiveSubject` overwriting all methods except one. Therefore mentioned method (`set(Map<String,Object)`) didn't activated the if statement

**Comment by nbrugger-tgm** [comment]
### fix merged

the fix will be shipped version `3.1.0b2`
